python/pyarrow/_dataset.pyx

```diff
@@ -76,8 +76,54 @@ cdef class FileFormat:
 
 cdef class ParquetFileFormat(FileFormat):
 
+    cdef:
+        CParquetFileFormat* parquet_format
+
     def __init__(self):
-        self.init(shared_ptr[CFileFormat](new CParquetFileFormat()))
+        self.init(<shared_ptr[CFileFormat]> CParquetFileFormat.Make())
+        self.parquet_format = <CParquetFileFormat*> self.wrapped.get()
+
+    @property
+    def use_buffered_stream(self):
+        """Read files through buffered input streams rather than
+        loading entire chunks at a time."""
+        return self.parquet_format.use_buffered_stream
+
+    @use_buffered_stream.setter
+    def use_buffered_stream(self, bint value):
+        self.parquet_format.use_buffered_stream = value
+
+    @property
+    def buffer_size(self):
+        """Size of buffered stream, if enabled."""
+        return self.parquet_format.buffer_size
+
+    @buffer_size.setter
+    def buffer_size(self, int value):
+        self.parquet_format.buffer_size = value
+
+    @property
+    def read_dict_indices(self):
+        """Indices of columns which should be read as dictionaries."""
+        return self.parquet_format.read_dict_indices
+
+    @read_dict_indices.setter
+    def read_dict_indices(self, set values):
+        self.parquet_format.read_dict_indices.clear()
+        for value in values:
+            self.read_dict_index(int(value))
+
+    def read_dict_index(self, int value):
```

Comment:
This method maybe doesn't need to be public?

Reply:
I thought that users might want to incrementally specify columns, in which case setting all of dict indices at once would be annoying. I can definitely remove it if you don't think it's necessary